core/src/main/java/org/fao/geonet/kernel/datamanager/base/BaseMetadataManager.java

```diff
@@ -1350,4 +1352,23 @@ boolean hasReferencingMetadata(ServiceContext context, AbstractMetadata metadata
         return this.searchManager.query(query.toString(), null, 0, 0).hits().total().value() > 0;
     }
 
+    private void indexMdsReferencingSubTemplate(ServiceContext context, AbstractMetadata subTemplate) throws Exception {
+        String query = String.format("xlink:*%s*", subTemplate.getUuid());
+        SearchResponse response = this.searchManager.query(query, null, Set.of("id"),0, maxMdsReferencingSubTemplate);
+        if (response.hits().total().value() > maxMdsReferencingSubTemplate) {
+            LOGGER_DATA_MANAGER.warn("Too many mds referencing subtemplate {}. Max allowed is {}.", subTemplate.getUuid(), maxMdsReferencingSubTemplate);
+            throw new TooManyMDsForThisSubTemplateException("To avoid running out of resources, this exception is thrown when the maximum number of subtemplates referencing a metadata record has been reached (" + maxMdsReferencingSubTemplate + ").");
+        }
+        Stream<Hit<?>> hits = response.hits().hits().stream();
+        List<String> toIndex = hits
+            .map(Hit::source)
+            .filter(ObjectNode.class::isInstance)
+            .map(ObjectNode.class::cast)
+            .map(x -> x.get("id"))
+            .filter(Objects::nonNull)
+            .map(JsonNode::asText)
+            .collect(Collectors.toList());
+
+        metadataIndexer.batchIndexInThreadPool(context, toIndex);
```

Comment:
There is a trouble, i.e. there are cases where indexation starts before subtemplate hibernate commit has completed.